### ISSUE TITLE
Default batch size is incorrect

### ISSUE THREAD

**Issue by anderspitman** [mongodb-labs/mongo-rust-driver-prototype#209]
Using this driver for a basic query, the results are being returned to my client much slower than doing the same query with `pymongo`. My best guess is that the rust driver is making a bunch of round trips to the database to retrieve the results a bit at a time, while `pymongo` gets everything in one shot. Again, this is just a guess. If I'm right is there a way to change the behavior. Or is it something else altogether?

**Comment by saghm** [comment]
Can you give an example of the query you're using? It's hard to tell what might be going on without seeing the code that's running

**Comment by anderspitman** [comment]
```rust
    let doc = doc! {
        "sensor_id" => 1
    };

    let cursor = collection.find(Some(doc.clone()), None)

    for result in cursor {
        if let Ok(item) = result {
            println!("{}", item);
        }
    }
}
```

That returns about 1500 records. Even if I remove the print it takes a couple seconds to loop through all of them. The find call itself is basically instantaneous.

**Comment by anderspitman** [comment]
Sample results:
```
{ _id: ObjectId("590a6e6d2c015b3e71f76411"), sensor_id: 1, temperature_celsius: 30, humidity: 32, datetime: "2017-05-03T16:57:33.385399" }
{ _id: ObjectId("590a6e6e2c015b3e71f76412"), sensor_id: 4, temperature_celsius: 27, humidity: 17, datetime: "2017-05-03T16:57:34.058113" }
{ _id: ObjectId("590a6e6f2c015b3e71f76413"), sensor_id: 7, temperature_celsius: 32, humidity: 15, datetime: "2017-05-03T16:57:35.220286" }
{ _id: ObjectId("590a6e732c015b3e71f76414"), sensor_id: 5, temperature_celsius: 31, humidity: 15, datetime: "2017-05-03T16:57:39.307786" }
```

**Comment by anderspitman** [comment]
Quick and dirty comparison with pymongo. Times in seconds.

Time to execute `find`:
pymongo: 0.000126
rust mongodb: 0.048850

Time to loop through results:
pymongo: 0.057684
rust mongodb:  3.587362

**Comment by saghm** [comment]
I think your instinct is correct; the default batch size for the Rust driver is 20 documents, whereas PyMongo doesn't seem specify one, and [the server defaults to 101](https://docs.mongodb.com/manual/tutorial/iterate-a-cursor/#cursor-batches). This looks like it might be something we need to change, but in the meantime, can you try [passing in a custom batch size](https://docs.rs/mongodb/0.2.7/mongodb/coll/options/struct.FindOptions.html#structfield.batch_size)?

@kyeah, any idea why we have the default batch size as 20? From looking back over the git history, I seem to be the one who introduced it (as a hardcoded value, not as a constant), but I can't find where I got that number from (maybe from seeing the shell show 20 documents by default at a time)? I think we should probably be defaulting to 0, which lets the server decide.

**Comment by anderspitman** [comment]
That seems to do the trick.

`batch_size = 101` (mongodb server default):
Time to execute `find`:
0.069879
Time to loop through results:
0.983068

`batch_size = 2000` (big enough for the entire result; maybe same behavior as pymongo?):
Time to execute `find`:
0.294689
Time to loop through results:
0.002365

**Comment by anderspitman** [comment]
Oh, I just read your comment more carefully. I guess the more useful test is setting batch_size to 0:
`batch_size = 101` (mongodb server chooses):
Time to execute find:
0.069820
Time to loop through results:
 0.299829

**Comment by kyeah** [comment]
@saghm yep, i assume this comes from the mongoShell default. I'm +1 to using 0, which should act the same as other drivers in normal cases.

**Comment by saghm** [comment]
Awesome. I'll open a PR with this change sometime tonight.

@anderspitman, thanks for noticing this issue!

**Comment by saghm** [comment]
@anderspitman, we just released 0.2.8, which fixes the default batch size (by letting the server decide it). Feel free to try it out, and let us know if there are any issues!
